ggshield/utils/files.py

```diff
@@ -55,17 +62,19 @@ def get_filepaths(
             if not recursive:
                 raise UnexpectedDirectoryError(path)
 
-            if not ignore_git and is_git_dir(path):
-                target_filepaths = (
-                    get_filepaths_from_ref("HEAD", wd=path)
-                    if ignore_git_staged
-                    else git_ls(path)
-                )
-                _targets = {path / x for x in target_filepaths}
-                if include_git_unstaged:
-                    _targets.update({path / x for x in git_ls_unstaged(path)})
-            else:
-                _targets = path.rglob(r"*")
+            _targets = set()
+            if list_files_mode != ListFilesMode.FILES_ONLY:
+                if is_git_dir(path) and not list_files_mode == ListFilesMode.ALL:
```

Comment:
I'd rewrite this as:

```
if list_files_mode != ListFilesMode.ALL and is_git_dir(path):
```

The reason for this is that it removes one place where git is required. When installing ggshield in an environment without git (for example a docker container), we should be able to run `ggshield secret scan path`.